packages/web3-eth/src/web3_eth.ts

```diff
@@ -238,6 +242,99 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 		return rpcMethodsWrappers.getGasPrice(this, returnFormat);
 	}
 
+	/**
+	 * @param returnFormat ({@link DataFormat} defaults to {@link DEFAULT_RETURN_FORMAT}) Specifies how the return data should be formatted.
+	 * @returns the current maxPriorityFeePerGas per gas in wei.
+	 *
+	 * ```ts
+	 * web3.eth.getMaxPriorityFeePerGas().then(console.log);
+	 * > 20000000000n
+	 *
+	 * web3.eth.getMaxPriorityFeePerGas({ number: FMT_NUMBER.HEX , bytes: FMT_BYTES.HEX }).then(console.log);
+	 * > "0x4a817c800"
+	 * ```
+	 */
+	public async getMaxPriorityFeePerGas<
```

Comment:
will be great to add e2e tests for this as well for geth.

Reply:
Will be added in: https://github.com/web3/web3.js/pull/6748